code/modules/power/lighting.dm

```diff
@@ -19,14 +27,24 @@
 	plane = MOB_PLANE
 	layer = ABOVE_MOB_LAYER
 	var/stage = 1
-	var/fixture_type = "tube"
+	var/fixture_type = /obj/machinery/light
 	var/sheets_refunded = 2
-	var/obj/machinery/light/newlight = null
 
-/obj/machinery/light_construct/New()
-	..()
-	if (fixture_type == "bulb")
-		icon_state = "bulb-construct-stage1"
+/obj/machinery/light_construct/New(atom/newloc, obj/machinery/light/fixture = null)
+	..(newloc)
+	if(fixture)
+		fixture_type = fixture.type
+		fixture.transfer_fingerprints_to(src)
+		set_dir(fixture.dir)
+		stage = 2
+	update_icon()
+
+/obj/machinery/light_construct/update_icon()
+	switch(stage)
+		if(1) icon_state = "tube-construct-stage1"
+		if(2) icon_state = "tube-construct-stage2"

```

Comment:
Please put the contents of an if/else/for check under the check itself, for readability.